paginator/paginator_test.go

```diff
@@ -32,6 +38,48 @@ type item struct {
 	Order   Order   `gorm:"foreignKey:OrderID"`
 }
 
+/* JSON type taken from https://gorm.io/docs/data_types.html */
+
+type JSON struct {
+	json.RawMessage
+}
+
+// Scan scan value into Jsonb, implements sql.Scanner interface
+func (j *JSON) Scan(value interface{}) error {
+	bytes, ok := value.([]byte)
+	if !ok {
+		return errors.New(fmt.Sprint("Failed to unmarshal JSONB value:", value))
+	}
+
+	result := json.RawMessage{}
+	err := json.Unmarshal(bytes, &result)
+	*j = JSON{result}
+	return err
+}
+
+// Value return json value, implement driver.Valuer interface
+func (j JSON) Value() (driver.Value, error) {
+	return j.MarshalJSON()
+}
+
+func (j JSON) GetCustomTypeValue(meta interface{}) interface{} {
+	// The decoding & mapping logic should be implemented by custom type provider
+	d := map[string]interface{}{}
+	err := json.Unmarshal(j.RawMessage, &d)
+	if err != nil {
+		panic("TODO")
```

Comment:
Just coming across this line and pondering about: 

Could it be better to add another error return to `GetCustomTypeValue` signature, to let user be able to handle error when getting custom value fails? I would like to hear your opinion about it 🙂

```go
type CustomType interface {
	GetCustomTypeValue(meta interface{}) (interface{}, error)
}
```

Reply:
Great idea. Yes, I think it would be better, as doing so we would allow to error gracefully instead of panicking if there are issues in `GetCustomTypeValue` code.

I added error handling now.